paasta_tools/autoscaling_lib.py

```diff
@@ -361,27 +363,36 @@ def humanize_error(error):
         return 'utilization within thresholds'
 
 
-def autoscale_services(soa_dir=DEFAULT_SOA_DIR):
-    try:
-        with create_autoscaling_lock():
-            cluster = load_system_paasta_config().get_cluster()
-            services = get_services_for_cluster(
+def get_configs_of_services_to_scale(cluster, soa_dir=DEFAULT_SOA_DIR):
+    services = get_services_for_cluster(
+        cluster=cluster,
+        instance_type='marathon',
+        soa_dir=soa_dir,
+    )
+    configs = []
+    for service, instance in services:
+        try:
+            service_config = load_marathon_service_config(
+                service=service,
+                instance=instance,
                 cluster=cluster,
-                instance_type='marathon',
                 soa_dir=soa_dir,
             )
-            configs = []
-            for service, instance in services:
-                service_config = load_marathon_service_config(
-                    service=service,
-                    instance=instance,
-                    cluster=cluster,
-                    soa_dir=soa_dir,
-                )
-                if service_config.get_max_instances() and service_config.get_desired_state() == 'start' \
-                        and service_config.get_autoscaling_params()['decision_policy'] != 'bespoke':
-                    configs.append(service_config)
+            if service_config.get_max_instances() and service_config.get_desired_state() == 'start' \
+                    and service_config.get_autoscaling_params()['decision_policy'] != 'bespoke':
+                configs.append(service_config)
+        except NoDeploymentsAvailable:
```

Comment:
can you move this catch to line 381 to be clear it's ``load_marathon_service_config`` that'll throw the exception (and add a ``continue`` after the debug line to skip to the next iteration)